k4FWCore/src/PodioDataSvc.cpp

```diff
@@ -52,21 +53,27 @@ StatusCode PodioDataSvc::finalize() {
 }
 
 StatusCode PodioDataSvc::clearStore() {
-  for (auto& collNamePair : m_collections) {
-    if (collNamePair.second != nullptr) {
-      collNamePair.second->clear();
-    }
+  // as the frame takes care of the ownership of the podio::Collections,
```

Comment:
Actually the creation is done just a few lines further below.

Reply:
Yes, I saw that, I was more wondering whether it would be possible to split the clean up (at the end of an event) and the creation of new content at the beginning of the next event. The main potential issue I see is that it will be very hard to signal to the next event that there were no events left from the reader, but if we were at the beginning of an event, we could rather easily abort it.

Reply:
That is a problem in the design here. One of the few places where the state machine is not nicely done. And the check whether there is nothing left is triggered in PodioInput (where it eventually should be anyhow).

Reply:
Thanks for the clarification. In that case, we will indeed address this when we make the PodioInput handle everything.